<?xml version="1.0"?>

<robot name="xd" xmlns:xacro="http://www.ros.org/wiki/xacro">
    <xacro:property name="params" value="${load_yaml('../param/params.yaml')}"/>
    <link name="base_link">
        <visual>
            <geometry>
                <cylinder length="0.3" radius="0.2"/>
            </geometry>
        <origin xyz="0 0 0.15" rpy="0 0 0"/>
        </visual>
    </link>
    <link name="link0"/>
    <link name="link1"/>
    <link name="link2">
        <visual>
            <geometry>
                <box size="${params['a2']} 0.1 0.1"/>
            </geometry>
            <origin xyz="${params['a2']/2} 0 0" rpy="0 0 0"/>
        </visual>
    </link>
    <link name="manipulator">
        <visual>
            <geometry>
                <box size="0.1 0.1 0.1"/>
            </geometry>
            <origin xyz="0.05 0 0" rpy="0 0 0"/>
        </visual>
    </link>
    <joint name="joint0" type="fixed">
        <parent link="base_link"/>
        <child link="link0"/>
        <origin xyz="0 0 0.3" rpy="0 0 0"/>
    </joint>
    <joint name="joint1" type="continuous">
        <parent link="link0"/>
        <child link="link1"/>
        <origin xyz="0 0 0" rpy="0 0 0"/>
        <axis xyz="0 0 1"/>
    </joint>
    <joint name="joint2" type="continuous">
        <parent link="link1"/>
        <child link="link2"/>
        <origin xyz="0 0 0" rpy="-1.57 0 0"/>
        <axis xyz="0 0 1"/>
    </joint>
    <joint name="joint3" type="continuous">
        <parent link="link2"/>
        <child link="manipulator"/>
        <origin xyz="0.5 0 0" rpy="0 0 0"/>
        <axis xyz="0 0 1"/>
    </joint>

</robot>
